### PR TITLE
Fixing Oculus hands rendering broken by recent WebXR Hands API change

### LINKED ISSUE #21666: VR - Oculus Quest 2 hand-bone orientation changes
While working on the Oculus Quest 2, we have just found that the bone positions given by XRJoint are no longer correctly mapping to the mesh.

**To Reproduce**
1. Use Oculus Quest 2
2. Go to https://threejs.org/examples/?q=vr#webxr_vr_handinput_profiles
3. View hands with hand tracking

**Video**
https://user-images.githubusercontent.com/33930602/115038901-41d81b00-9ed0-11eb-8486-5de53a566b0c.mp4

**Screenshot**
image

**Expected behavior**
Hand models shouldn't be warped.

**Quick Fix**
XRHandOculusMeshModel.js

https://github.com/mrdoob/three.js/blob/f2086ff617a42a40c57977c06ab81544e9d4b648/examples/jsm/webxr/XRHandOculusMeshModel.js#L125

```js
bone.position.copy( position.clone().multiplyScalar( 100 ) );
bone.quaternion.copy( XRJoint.quaternion );
// bone.scale.setScalar( XRJoint.jointRadius || defaultRadius );

if(this.handedness === 'left') {
	bone.rotateY(Math.PI/2);
	bone.rotateX(Math.PI);

} else {
	bone.rotateY(-Math.PI/2);
}
```

**Platform:**
- OculusBrowser/15.1.0.2.46.283931095 SamsungBrowser/4.0 Chrome/89.0.4389.105
 - Device: [Oculus Quest 2]
 - Three.js version: [r125.2, r127]

### PR DESCRIPTION
**Pull request by felixtrz**
Fixing oculus hands broken by recent WebXR Hands API change by Oculus Browser.
Disabling low poly option in handinput - profile example until a up-to-date low poly model is available

Fixed  #21666.

### CODE REVIEW

**Comment by Mugen87** [comment]
I guess that would fix #21666, right?

**Comment by mrdoob** [comment]
Nice! Would it be possible to make glb files instead of fbx though?

**Comment by fe1ixz** [comment]
> I guess that would fix #21666, right?

Yes it will

**Comment by fe1ixz** [comment]
> Nice! Would it be possible to make glb files instead of fbx though?

We do have glb files, I kept the fbx because the hand model files in the repo were all fbx. Let me change that, it'll be a quick fix

File: examples/jsm/webxr/Text2D.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import * as THREE from "../../../build/three.module.js";
+
+function createText(message, height) {
+  var canvas = document.createElement("canvas");
+  var context = canvas.getContext("2d");
+  var metrics = null,
+    textHeight = 100,
+    textWidth = 100,
+    actualFontSize = 2;
```
Comment (Mugen87) [inline examples/jsm/webxr/Text2D.js:9]: It seems `actualFontSize` is never used.

File: examples/jsm/webxr/Text2D.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import * as THREE from "../../../build/three.module.js";
+
+function createText(message, height) {
+  var canvas = document.createElement("canvas");
+  var context = canvas.getContext("2d");
+  var metrics = null,
+    textHeight = 100,
+    textWidth = 100,
```
Comment (Mugen87) [inline examples/jsm/webxr/Text2D.js:8]: Since you declare `textWidth` in line `12`, this one can be removed.

File: examples/jsm/webxr/Text2D.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import * as THREE from "../../../build/three.module.js";
+
+function createText(message, height) {
+  var canvas = document.createElement("canvas");
```
Comment (Mugen87) [inline examples/jsm/webxr/Text2D.js:4]: Do you mind using `let`/`const` in this file?

Comment (fe1ixz) [reply examples/jsm/webxr/Text2D.js]: Fixed!

**Review by Mugen87 — DISMISSED** [review]

**Comment by Mugen87** [comment]
https://raw.githack.com/fe1ixz/three.js/unbreakhands/examples/index.html?q=webxr

**Comment by Mugen87** [comment]
Tested the new examples on a Quest 2 today. Really awesome!

File: examples/jsm/webxr/OculusHandModel.js [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+class XRHandOculusMeshModel {
+  constructor(handModel, controller, handedness) {
+    this.controller = controller;
+    this.handModel = handModel;
+
+    this.bones = [];
+    const loader = new FBXLoader();
+
+    // loader.setPath("");
+    loader.load("models/fbx/OculusHandNew_" + handedness + ".fbx", (object) => {
```
Comment (Mugen87) [inline examples/jsm/webxr/OculusHandModel.js:43]: Is it possible to use `GLTFLoader` here, too? It seems the FBX assets can then be removed.

File: examples/jsm/webxr/OculusHandModel.js [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    });
+
+    controller.addEventListener("disconnected", () => {
+      this.clear();
+      this.motionController = null;
+    })
+  }
+
+  updateMatrixWorld(force) {
+    Object3D.prototype.updateMatrixWorld.call(this, force);
```
Comment (Mugen87) [inline examples/jsm/webxr/OculusHandModel.js:163]:
```js
super. updateMatrixWorld( force );
```

File: examples/jsm/webxr/OculusHandModel.js [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+  "pinky-finger-metacarpal",
+  "pinky-finger-phalanx-proximal",
+  "pinky-finger-phalanx-intermediate",
+  "pinky-finger-phalanx-distal",
+  "pinky-finger-tip",
+];
+const TOUCH_RADIUS = 0.01;
+const POINTING_JOINT = "index-finger-tip";
+
+class XRHandOculusMeshModel {
```
Comment (Mugen87) [inline examples/jsm/webxr/OculusHandModel.js:34]: Do you mind explaining why `XRHandOculusMeshModel` is defined at this place again? There is already `examples/jsm/webxr/XRHandOculusMeshModel.js`.

**Comment by fe1ixz** [comment]
1. Updated existing and new samples to fetch models from webxr input profile cdn
2. Removed old fbx and glb models
3. Removed redundant implementation of XRHandMeshModel

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
Bug fix: https://github.com/mrdoob/three.js/commit/a92265f02be7c5b4f7abdbe05b1f21908485a845

**Comment by mrdoob** [comment]
Minor: https://github.com/mrdoob/three.js/commit/9bbd837deb3e1d43d5c6c7054211eda5864bea06

**Comment by mrdoob** [comment]
Bug fix: https://github.com/mrdoob/three.js/commit/2c9b14387fb40d4ef1dc7aa06fa39d1515cc8433

**Comment by mrdoob** [comment]
Seems like the code had not been tested.
I've fixed it and made sure it works on the Quest: https://github.com/mrdoob/three.js/commit/571347f504dec332f6905bb3d8ef43f1b405d45c

File: examples/jsm/webxr/XRHandModelFactory.js [hunk 7]
```diff
@@ -77,7 +84,15 @@ class XRHandModelFactory {
...
-					handModel.motionController = new XRHandOculusMeshModel( handModel, controller, this.path, xrInputSource.handedness, options );
+					fetchProfile(xrInputSource, DEFAULT_PROFILES_PATH, DEFAULT_PROFILE).then(({ profile, assetPath }) => {
+	
+						handModel.motionController = new XRHandMeshModel( handModel, controller, assetPath);
+
+					}).catch((err) => {
+
+						console.warn(err);
+
+					});
```
Comment (mrdoob) [inline examples/jsm/webxr/XRHandModelFactory.js:95]: @fe1ixz I reverted this code. Why do we need to call `fetchProfile`? 🤔

**Comment by stewdio** [comment]
(You all are amazing! I am so excited to finally update [this old sign language demo from last summer](https://github.com/stewdio/handy.js) once these joint orientation fixes are out!)

**Comment by stewdio** [comment]
Slight tangent: I saw the `Text2D` file now included in the `WebXR` folder. Would something like **SurfaceText** be of interest? If so, I could clean it up and submit. https://github.com/stewdio/handy.js/blob/master/vendor/Moar/scripts/SurfaceText.js I even have a version kicking around somewhere that handles line wraps.

**Pull request merged**